lib/badge.rb

```diff
@@ -0,0 +1,23 @@
+# Class reprensenting a badge a player can earn
+class Badge
+  attr_reader :short, :title, :description
+
+  def initialize(short, title, description, &block)
+    @short = short
+    @title = title
+    @description = description
+    @block = block
+  end
+
+  def earned_by?(player)
+    # Return false if no evaluation predicate has been provided
+    return false if !@block || @block.call(player) == false
+    @block.call(player) == true || @block.call(player) > 0
```

Comment:
Does that mean `@block` is called 3 times when it returns an `int` ?
Since the block is interating on all cotnributions, for each player, it may kill response time (which is already quite high) if there are many players, with many contributions.
It should probably be called once and result stored in a variable at the method beginning.

Reply:
Yep my mistake, sorry. I forgot to check performance before making things just to work